airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/config/SyncBeanFactory.kt

```diff
@@ -179,4 +190,32 @@ class SyncBeanFactory {
     @Singleton
     @Named("defaultDestinationTaskLauncherHasThrown")
     fun defaultDestinationTaskLauncherHasThrown(): AtomicBoolean = AtomicBoolean(false)
+
+    /* *****
+     * CHECK
+     * *****/
+
+    // TODO combine this with the actual catalog factory method? where even is that method
+    @Singleton
+    @Requires(property = Operation.PROPERTY, value = "check")
```

Comment:
Odd to put something that exists exclusively for CHECK in the `Sync` bean factory imho. As for the requires bit, it might be better for the bean logic to take the `Operation.PROPERTY` as an arg and conditionally return the correct catalog implementation.

Reply:
how does https://github.com/airbytehq/airbyte/pull/59691/commits/aa0a9908a831f7ea1639b80a9b24b16243f5a5bc look?